MaintenanceManager/MaintenanceManager.cpp

```diff
@@ -1106,9 +1124,11 @@ namespace WPEFramework {
 
                 std::lock_guard<std::mutex> guard(m_callMutex);
 
+		rdkvfwrfc = checkRdkvFWFlag();
                 /* check if maintenance is on progress or not */
                 /* if in progress restrict the same */
-                if ( MAINTENANCE_STARTED != m_notify_status ){
+		/* This feature is only applicable when firmware download rfc is true */
+                if ( (rdkvfwrfc == true) || (MAINTENANCE_STARTED != m_notify_status) ){
```

Comment:
Line 1135 can be moved before lock_guard as its just validation of input params.

Reply:
Ok.